### PR TITLE
DOCS-2631: Add dataset export option CLI

### PR DESCRIPTION
**Pull request by sguequierre**
* adds dataset export option

attempted to add explanation and usage for the flags which meant formatting a little differently in the table, lmk and I can remove

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -340,6 +341,7 @@ done
 | `database hostname` | Get the MongoDB Atlas Data Federation instance hostname and connection URI. See [Configure data query](/use-cases/sensor-data-query-with-third-party-tools/#configure-data-query) | - |
 | `dataset add` | Add a new image to an existing dataset by its file id, or add a group of images by specifying a filter | `filter` |
 | `dataset remove` | Remove an existing image from a dataset by its file id, or remove a group of images by specifying a filter | `filter` |
+| `dataset export` | Download all the data from a dataset | `dataFlagDestination`: Output directory for downloaded data. <br> `datasetFlagDatasetID`: Dataset ID of the dataset to download. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID**. <br> `datasetFlagIncludeJSONLines`: Set to `true` to include JSON Lines files for local testing. *(optional)* <br> `dataFlagParallelDownloads`: Number of download requests to make in parallel. *(optional)* Default: `100` |
```
Comment (npentrel) [inline docs/cli.md:344]: Can you try this with html
```suggestion
| `dataset export` | Download all the data from a dataset | <ul><li>`dataFlagDestination`: Output directory for downloaded data.</li><li>`datasetFlagDatasetID`: Dataset ID of the dataset to download. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID**.</li><li>`datasetFlagIncludeJSONLines`: Set to `true` to include JSON Lines files for local testing. *(optional)*</li><li>`dataFlagParallelDownloads`: Number of download requests to make in parallel. *(optional)* Default: `100`</li></ul> |
```

File: docs/cli.md [hunk 2]
```diff
@@ -363,7 +365,7 @@ done
 | `--component-name` | Filter by specified component name |`export`, `delete`| Optional |
 | `--component-type` | Filter by specified component type |`export`, `delete`| Optional |
 | `--component-model` | Filter by specified component model |`export`, `delete`| Optional |
-| `--dataset-id` | Dataset to add or remove images from |`dataset`| **Required** |
+| `--dataset-id` | Dataset to add or remove images from | `dataset` | **Required** |
```
Comment (sguequierre) [inline docs/cli.md:368]:
```suggestion
| `--dataset-id` | Dataset to add or remove images from |`dataset`| **Required** |
```

File: docs/cli.md [hunk 3]
```diff
@@ -296,6 +296,7 @@ viam data export --destination=<output path> --data-type=<output data type> [...
 viam data delete [...named args]
 viam data dataset add --dataset-id=<dataset-id> --file-ids=<file-id-or-ids> --location-id=<location-id> --org-id=<org-id> [...named args]
 viam data dataset remove --dataset-id=<dataset-id> --file-ids=<file-id-or-ids> --location-id=<location-id> --org-id=<org-id> [...named args]
+viam data dataset export --destination=<output-directory> --datasetFlagDatasetID=<dataset-id>
```
Comment (sguequierre) [inline docs/cli.md:299]: destination already included export in "named arguments" @tahiyasalam LMK if this needs correcting, a bit confused about flags not being renamed to simple names like destination is

Comment (npentrel) [reply docs/cli.md:299]: Please make sure you install the latest version of the CLI and check it to get this information:

```
(viam-docs) ➜  viam-docs git:(viam-agent-full) ✗ viam dataset export
NAME:
   viam dataset export - download data from a dataset

USAGE:
   viam dataset export --destination=<destination> --dataset-id=<dataset-id> [other options] include-jsonl parallel

OPTIONS:
   --dataset-id value   dataset ID of the dataset to be downloaded
   --destination value  output directory for downloaded data
   --include-jsonl      option to include JSON Lines files for local testing (default: false)
   --parallel value     number of download requests to make in parallel (default: 100)

Error: Required flags "destination, dataset-id" not set
```

Alternatively, the information is also in [app.go](https://github.com/tahiyasalam/rdk/blob/main/cli/app.go#L686). The lines that set the name parameters are:
- https://github.com/tahiyasalam/rdk/blob/main/cli/dataset.go#L18
- https://github.com/tahiyasalam/rdk/blob/main/cli/app.go#L75
- etc.

You can search for the variable name and see where it is getting included to find out more.

File: docs/cli.md [hunk 4]
```diff
@@ -340,6 +341,7 @@ done
 | `database hostname` | Get the MongoDB Atlas Data Federation instance hostname and connection URI. See [Configure data query](/use-cases/sensor-data-query-with-third-party-tools/#configure-data-query) | - |
 | `dataset add` | Add a new image to an existing dataset by its file id, or add a group of images by specifying a filter | `filter` |
 | `dataset remove` | Remove an existing image from a dataset by its file id, or remove a group of images by specifying a filter | `filter` |
+| `dataset export` | Download all the data from a dataset | `datasetFlagDatasetID`: Dataset ID of the dataset to download. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID**.</li><li>`datasetFlagIncludeJSONLines`: Set to `true` to include JSON Lines files for local testing. _(optional)_</li><li>`dataFlagParallelDownloads`: Number of download requests to make in parallel. _(optional)_ Default: `100`</li></ul> |
```
Comment (npentrel) [inline docs/cli.md:344]: And now that I look at this again, these are named arguments that should go in the named arguments table below. not in this table.

File: docs/cli.md (hunk 3 above)
Comment (sguequierre) [reply docs/cli.md:299]: Thank you, I wasn't aware of this and was struggling to find. Will update!

File: docs/cli.md (hunk 4 above)
Comment (sguequierre) [reply docs/cli.md:344]: yup sounds good will fix, thanks naomi

File: docs/cli.md [hunk 5]
```diff
@@ -358,23 +360,24 @@ done
...
 | Argument | Description | Applicable commands | Required? |
 | -------- | ----------- | ------------------- | --------- |
-| `--destination` | Output directory for downloaded data |`export`| **Required** |
+| `--destination` | Output directory for downloaded data |`export`, `dataset export` | **Required** |
 | `--data-type` | Data type to be downloaded: either binary or tabular |`export`| **Required** |
 | `--component-name` | Filter by specified component name |`export`, `delete`| Optional |
 | `--component-type` | Filter by specified component type |`export`, `delete`| Optional |
 | `--component-model` | Filter by specified component model |`export`, `delete`| Optional |
-| `--dataset-id` | Dataset to add or remove images from |`dataset`| **Required** |
+| `--dataset-id` | Dataset to add or remove images from or download. To retrieve these IDs, navigate to your dataset’s page in the [Viam app](https://app.viam.com), click **…** in the left-hand menu, and click **Copy dataset ID**. | `dataset`, `dataset export` | **Required** |
```
Comment (npentrel) [inline docs/cli.md:368]: Nice

**Review by npentrel — APPROVED** [review]

**Review by tahiyasalam — APPROVED** [review]

**Pull request merged**